src/pinefarm/external/interface.py

```diff
@@ -67,6 +67,8 @@ def update_with_tmp(self):
     @staticmethod
     def install():
         """Install all needed programs."""
+        # Everybody needs LHAPDF unless explicitly skipped
+        _ = install.lhapdf()
```

Comment:
1) I completely agree that LHAPDF is (more or less) a hard requirement for `pinefarm`, but 2) I don't think this works
- can you remind me how method inheritance works in Python? i.e. although there is the `@staticmethod` decorator the method is inherited, right? Because there is no true "static" concept, but this is just an (ordinary) method of the class object, right? and so the method content is actually never triggered, because either the content is overloaded or never called - see also below
- actually the installing procedure is all here https://github.com/NNPDF/pinefarm/blob/install_lhapdf_for_all/src/pinefarm/cli/install.py and there is no "default" install routine that is always triggered, but you have to specifically request one; and then when you do so, the specific method of the external program is called (which at this point overwrote the parent)
- I think I like the current way, i.e. explicitly requiring an `pinefarm install` call - because otherwise `pinefarm` may install some arbitrarily complex code (more or less uncontrolled) on my system - a.k.a. lhapdf, mg5, matrix and what not
- moreover, I think at some point we intentionally delayed the import of `lhapdf` (as is currently in master), which would contradict your PR, no?

Reply:
Actually, after checking again - I believe the derived `install` methods are never called; maybe we should move some stuff from the `install` module to the respective `external.blub.install` methods

Reply:
> @staticmethod

I think you meant `abstract`. `static` just means that there's no `self` (which is perfectly ok for python). The method will be called unless overloaded and you can still do `super()`.

>  I believe the derived install methods are never called

They are by the respective programs at `run` time: https://github.com/NNPDF/pinefarm/blob/44e2f21f6655227b248284a9f74a1a8b4b14f55b/src/pinefarm/cli/run.py#L86

> I think I like the current way, 

This is still the current way, I haven't changed anything, only what happens in `install`.

>  I think at some point we intentionally delayed the import of lhapdf (as is currently in master), 

Yes, because only `run` needs it and a program can decide not to install `lhapdf` (by overloading the method) but at the moment all the runners need `lhapdf` unless the need for `lhapdf` is not overloaded as well.

Reply:
> They are by the respective programs at `run` time:
> 
> https://github.com/NNPDF/pinefarm/blob/44e2f21f6655227b248284a9f74a1a8b4b14f55b/src/pinefarm/cli/run.py#L86

I did not remember that ... and I'm inclined to revert that, i.e. to drop that line. I feel we should make installing (arbitrary) stuff a conscious decision and just cry if we can not run (maybe with a nice error message), i.e. I would do the opposite of this PR and make less automagic - but I'm open for discussion

Reply:
people may or actually in most cases may not like our install routines because they have special requirements/environments

Reply:
The way it was decided by common agreement (with which it seems we all disagree) it was that pinefarm would install its own things to its own environment `<some_config>/.prefix`. 
And then, before running the environment is updated to see that:
https://github.com/NNPDF/pinefarm/blob/44e2f21f6655227b248284a9f74a1a8b4b14f55b/src/pinefarm/install.py#L347

As it is right now in master, that is broken (this PR is a fixing a bug, it doesn't improve anything). It has been working "by chance" because `yadism` is pure python (so it was installed outside anyway) and the others don't rely in python (the pkg-config / LD_LIBRARY is ok). It has failed for me today because I happened to try to run it in a system with no lhapdf installed at all.

Now, I am happy for you to remove that line. You need to make sure then that the different runners fail early with what's missing. Probably changing the `.install()` line to `.check_missing()` or something.

(but I'd say that change is orthogonal to this PR)

Reply:
fine - I see you're point, you're just fixing what is already there ... let's discuss the strategy elsewhere

Reply:
(I'm totally on board with the removal of the automatic installation fwiw)